ament_tools/build_types/cmake.py

```diff
@@ -167,12 +167,27 @@ def _common_cmake_on_build(self, should_run_configure, context, prefix, extra_cm
             solution_file = solution_file_exists_at(
                 context.build_space, context.package_manifest.name)
             cmd = prefix + [MSBUILD_EXECUTABLE]
-            if '-j1' in context.make_flags:
-                cmd += ['/m']
+            env = None
+            # Convert make parallelism flags into msbuild flags
+            msbuild_flags = [
+                x.replace('-j', '/m:') for x in context.make_flags if x.startswith('-j')
+            ]
+            if msbuild_flags:
+                cmd += msbuild_flags
+                # If there is a parallelism flag in the msbuild_flags and it is not /m1,
+                # then turn on /MP for the compiler too.
+                if any(['/m' in x for x in msbuild_flags]) and '/m:1' not in msbuild_flags:
+                    env = dict(os.environ)
+                    if 'CL' in env:
+                        # make sure there isn't anything in there trying to prevent MP
+                        if not any(['/MP' in x for x in env['CL']]):
```

Comment:
`in` is not a very defensive check. The substring could be anywhere. Maybe check for `startswith` instead.